src/HealthGPS/kevin_hall_model.cpp

```diff
@@ -168,18 +221,23 @@ void KevinHallModel::set_nutrient_intakes(Person &person) const {
 }
 
 void KevinHallModel::initialise_energy_intake(Person &person) const {
-    set_energy_intake(person);
 
-    // Begin at steady state (EI = EE).
-    person.risk_factors["EnergyExpenditure"_id] = person.risk_factors.at("EnergyIntake"_id);
+    // Initialise energy intake.
+    set_energy_intake(person);
 
-    // TODO: set old value to new value
+    // Start with previous = current.
+    double energy_intake = person.risk_factors.at("EnergyIntake"_id);
+    person.risk_factors["EnergyIntake_previous"_id] = energy_intake;
 }
 
 void KevinHallModel::update_energy_intake(Person &person) const {
-    set_energy_intake(person);
 
-    // TODO: set old value to previous value
+    // Set previous energy intake.
+    double previous_energy_intake = person.risk_factors.at("EnergyIntake"_id);
+    person.risk_factors.at("EnergyIntake_previous"_id) = previous_energy_intake;
+
+    // Update energy intake.
+    set_energy_intake(person);
 }
 
 void KevinHallModel::set_energy_intake(Person &person) const {
```

Comment:
Pedantic point, but I feel like the `set_` prefix makes it sound like more of a "setter" (i.e. for setting the intake to a specified number). How about `calculate_`/`recalculate_`?

Reply:
That's what these `set_` functions are essentially doing: setting risk factors in people. To me, `calculate_` implies returning that value to caller.

Reply:
Although maybe it's worth it for consistency. I'll change them in the next PR.

Reply:
Up to you. I agree that both naming schemes have problems :shrug: